lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -723,8 +669,126 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     end
   end
 
+  defp do_accept_transaction(%Transaction{type: :contract, data: %TransactionData{code: code}}, _) do
+    if byte_size(code) == 0 do
+      {:error, "invalid contract type transaction -  code is empty"}
+    else
+      :ok
+    end
+  end
+
+  defp do_accept_transaction(
+         %Transaction{type: :data, data: %TransactionData{content: "", ownerships: []}},
+         _
+       ),
+       do: {:error, "invalid data type transaction - Both content & ownership are empty"}
+
+  defp do_accept_transaction(%Transaction{type: :data}, _), do: :ok
```

Comment:
I think this match don't have more value than the one after which match all remaining cases